core/store/src/trie/mem/arena/alloc.rs

```diff
@@ -68,6 +70,14 @@ impl Allocator {
         }
     }
 
+    /// Adds a new page to the arena, and updates the next_ptr to the beginning of
+    /// the new page.
+    fn new_page(&mut self, arena: &mut ArenaMemory) {
+        arena.chunks.push(vec![0; CHUNK_SIZE]);
+        self.next_ptr = ArenaPos { chunk: (arena.chunks.len() - 1) as u32, pos: 0 };
```

Comment:
Use `u32::try_from`. Although right now with `CHUNK_SIZE` at 4MiB it would require 18PiB of memory to overflow this number, there isn’t much preventing somebody from reducing the chunk size without taking note of this conversion.

Reply:
Additionally, it isn’t clear to me what would happen if `self.next_ptr` is not invalid at this point in time? Would it just leak?

Might it be better to move the conditions for `is_invalid` and `arena.chunks[self.next_ptr.chunk as usize].len() <= self.next_ptr.pos as usize + allocation_size` if only to make sure handling of these conditions remains correct by any potential future callers?

Reply:
Oh I was looking for a safer conversion function! Thanks!

Fixed the new_page -> new_chunk.